### PR TITLE
lucuma-itc-testkit

### PR DESCRIPTION
**Pull request by swalker2m**
Adds a `testkit` module with arbitraries for the client input classes.

### CODE REVIEW

**Comment by swalker2m** [comment]
This PR introduces the `testkit` but also seems to want to download it.  What am I doing wrong?

```
[error] sbt.librarymanagement.ResolveException: Error downloading edu.gemini:lucuma-itc-testkit_sjs1_3:0.6.0
[33](https://github.com/gemini-hlsw/lucuma-itc/actions/runs/4226072980/jobs/7339115140#step:10:34)
[error]   Not found
[34](https://github.com/gemini-hlsw/lucuma-itc/actions/runs/4226072980/jobs/7339115140#step:10:35)
[error]   Not found
[35](https://github.com/gemini-hlsw/lucuma-itc/actions/runs/4226072980/jobs/7339115140#step:10:36)
[error]   not found: /home/runner/.ivy2/localedu.gemini/lucuma-itc-testkit_sjs1_3/0.6.0/ivys/ivy.xml
[36](https://github.com/gemini-hlsw/lucuma-itc/actions/runs/4226072980/jobs/7339115140#step:10:37)
[error]   not found: https://repo1.maven.org/maven2/edu/gemini/lucuma-itc-testkit_sjs1_3/0.6.0/lucuma-itc-testkit_sjs1_3-0.6.0.pom
[37](https://github.com/gemini-hlsw/lucuma-itc/actions/runs/4226072980/jobs/7339115140#step:10:38)
[error]   download error: Caught java.io.IOException (Server returned HTTP response code: 504 for URL: https://s01.oss.sonatype.org/content/repositories/public/edu/gemini/lucuma-itc-testkit_sjs1_3/0.6.0/lucuma-itc-testkit_sjs1_3-0.6.0.pom) while downloading https://s01.oss.sonatype.org/content/repositories/public/edu/gemini/lucuma-itc-testkit_sjs1_3/0.6.0/lucuma-itc-testkit_sjs1_3-0.6.0.pom
[38](https://github.com/gemini-hlsw/lucuma-itc/actions/runs/4226072980/jobs/7339115140#step:10:39)
```

File: build.sbt [hunk 1]
```diff
@@ -208,17 +208,38 @@ lazy val benchmark = project
 
 val MUnitFramework = new TestFramework("munit.Framework")
 
+lazy val testkit = crossProject(JVMPlatform, JSPlatform)
+  .crossType(CrossType.Pure)
+  .in(file("modules/testkit"))
+  .dependsOn(client)
```
Comment (cquiroz) [inline build.sbt:214]: Mybe client depends on testkit?

**Review by cquiroz — APPROVED** [review]
Looks good to me

File: build.sbt (hunk 1 above)
Comment (swalker2m) [reply build.sbt:214]: Hmm, I guess I don't see the problem.

**Comment by toddburnside** [comment]
> This PR introduces the `testkit` but also seems to want to download it. What am I doing wrong?

This could just be a symptom of the problems sonatype is having today.

**Comment by swalker2m** [comment]
> This could just be a symptom of the problems sonatype is having today.

I was wondering if that might be the case, though it strikes me as odd that it even tries to download it.

**Comment by toddburnside** [comment]
> I was wondering if that might be the case, though it strikes me as odd that it even tries to download it.

The error I see when I look at the failed job is in the bincompat check. I think it has to download to check the compatibility? Obviously, there wouldn't be a previous version, but it doesn't know that.

**Comment by toddburnside** [comment]
> The error I see when I look at the failed job is in the bincompat check. I think it has to download to check the compatibility? Obviously, there wouldn't be a previous version, but it doesn't know that.

Carlos' PR in lucuma-core is having the same issue.

**Comment by swalker2m** [comment]
> Carlos' PR in lucuma-core is having the same issue.

:+1: Thanks.  I'll sit tight then.

**Pull request merged**